File: WorkoutTracker/src/app/account/models/user-details.ts
import { Gender } from "src/app/shared/models/gender";
import { ModelSize } from "../../shared/models/model-size";
import { ModelWeight } from "../../shared/models/model-weight";

export interface UserDetails {
    gender: Gender|null;
    height: ModelSize|null;
    weight: ModelWeight|null;
    dateOfBirth: Date|null;
    bodyFatPercentage: number|null;
}

File: WorkoutTrackerAPI/Data/Models/UserModels/UserDetails.cs
namespace WorkoutTrackerAPI.Data.Models.UserModels;

public class UserDetails : IDbModel
{
    public long Id { get; set; }
    public Gender? Gender { get; set; }
    public ModelSize? Height { get; set; }
    public ModelWeight? Weight { get; set; }
    public DateTime? DateOfBirth { get; set; }
    public double? BodyFatPercentage { get; set; }

    public string UserId { get; set; } = null!;
    public User? User { get; set; }
}

public enum Gender
{
    Male, Female
}
